### PR TITLE
[Feat 19] moonguard 0.1.0 docs

### LINKED ISSUE #19: [FEAT] MoonGuard Docs 0.1.0 Enhancement

### PR DESCRIPTION
**Pull request by AlexisTAE**
close #19

### CODE REVIEW

File: versioned_docs/version-0.1.0/larvis.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    public function register()
+{
+    if (! app()->environment('testing')) {
+        $this->reportable(function (Throwable $e) {
+            /** @var Larvis $larvis */
+            $larvis = app(Larvis::class);
+
+            $larvis->captureException($e);
+        });
+    }
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/larvis.md]: Format the code block with vs code.

File: versioned_docs/version-0.1.0/exceptions/exceptions.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+'exceptions' => [
+        'enabled' => true,
+        'notify_time_between_group_updates_in_minutes' => 15,
+        'exception_log' => [
+            'model' => \Taecontrol\MoonGuard\Models\ExceptionLog::class,
+        ],
+        'exception_log_group' => [
+            'model' => \Taecontrol\MoonGuard\Models\ExceptionLogGroup::class,
+        ],
+    ],
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/exceptions/exceptions.md]: Format the code block with vs code.

Use 
```php
[
  'exceptions' => [
  ]
]
```

File: versioned_docs/version-0.1.0/exceptions/exception-model.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+'exceptions' => [
+        'enabled' => true,
+        'notify_time_between_group_updates_in_minutes' => 15,
+        'exception_log' => [
+            'model' => \Taecontrol\MoonGuard\Models\ExceptionLog::class,
+        ],
+        'exception_log_group' => [
+            'model' => \Taecontrol\MoonGuard\Models\ExceptionLogGroup::class,
+        ],
+    ],
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/exceptions/exception-model.md]: Format the code block with vs code.

File: versioned_docs/version-0.1.0/notifications/notifications-mail.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+```php
+MAIL_MAILER=smtp
+MAIL_HOST=smtp.mailtrap.io
+MAIL_PORT=2525
+MAIL_USERNAME=xxxxxxxxxxx
+MAIL_PASSWORD=xxxxxxxxxxx
+MAIL_ENCRYPTION=tls
+MAIL_FROM_ADDRESS="demo@example.com"
+MAIL_FROM_NAME="${APP_NAME}"
+```
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/notifications/notifications-mail.md]: The description of this section should be directed to general purpose. Like MoonGuard needs the Mail configuration of your app env to send mail notifications.

File: versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-scheduling.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+     * Define the application's command schedule.
+     *
+     * @param  \Illuminate\Console\Scheduling\Schedule  $schedule
+     * @return void
+     */
+    protected function schedule(Schedule $schedule)
+    {
+        $schedule->command(CheckSslCertificateCommand::class)->everyTwoHours();
+    }
+```
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-scheduling.md]: Format the code block with vs code.

File: versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+Other behaviors of the SSL certificate can be disabled or modified in the configuration file.
+
+```php
+    'ssl_certificate_check' => [
+        'enabled' => true,
+        'model' => \Taecontrol\MoonGuard\Models\SslCertificateCheck::class,
+
+        'notify_expiring_soon_if_certificate_expires_within_days' => 7,
+        'cron_schedule' => '* * * * *',
+    ],
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate.md]: Format the code block with vs code.

File: versioned_docs/version-0.1.0/uptime/uptime-scheduling.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+     * Define the application's command schedule.
+     *
+     * @param  \Illuminate\Console\Scheduling\Schedule  $schedule
+     * @return void
+     */
+    protected function schedule(Schedule $schedule)
+    {
+                $schedule->command(CheckUptimeCommand::class)->everyMinute();
+    }
+```
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/uptime/uptime-scheduling.md]: Format the code block with vs code.

File: versioned_docs/version-0.1.0/site/site-model.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+| `exceptionLogs()`                                      | HasManyThrough | Returns a **HasManyThrough** relationship between the **Site** model and the **ExceptionLog** model through the **ExceptionLogGroup** model. This method allows you to retrieve the exception logs associated with the website. |
+| `exceptionLogGroups()`                                 | HasMany        | Returns a **HasMany** relationship between the **Site** model and the **ExceptionLogGroup** model. This method allows you to retrieve the exception log groups associated with the website.                                 |
+| `newCollection(array $models = [])`                    | SiteCollection | Overrides the **newCollection** method to return a custom **SiteCollection** instance. This method allows you to customize the collection returned when querying the **Site** model.                                        |
+
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/site/site-model.md:42]: Add new line.

File: versioned_docs/version-0.1.0/site/site-model.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+```php
+<?php
+
+use Illuminate\Database\Eloquent\Model;
+use Taecontrol\MoonGuard\Contracts\MoonGuardSite;
+
+class Site extends Model implements MoonGuardSite
+{
+    // Implementation of the interface
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/site/site-model.md]: Use "// Contract implementation" as comment

File: versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+
+use Illuminate\Database\Eloquent\Model;
+use Taecontrol\MoonGuard\Contracts\MoonGuardSslCertificateCheck;
+
+class UptimeCheck extends Model implements MoonGuardSslCertificateCheck
+{
+
+    // Implementation here
+
+    // Add other methods required by the interface
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md]: Use "// Contract implementation" as comment

File: versioned_docs/version-0.1.0/uptime/uptime-check-model.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+use Illuminate\Database\Eloquent\Model;
+use Taecontrol\MoonGuard\Contracts\MoonGuardUptimeCheck;
+
+class UptimeCheck extends Model implements MoonGuardUptimeCheck
+{
+
+    // Implementation here
+
+    // Add other methods required by the interface
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/uptime/uptime-check-model.md]: Use "// Contract implementation" as comment

File: versioned_docs/version-0.1.0/exceptions/exception-model.md [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+
+use Illuminate\Database\Eloquent\Model;
+use Taecontrol\MoonGuard\Contracts\MoonGuardExceptionLog;
+
+class UptimeCheck extends Model implements MoonGuardExceptionLog
+{
+
+    // Implementation here
+
+    // Add other methods required by the interface
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/exceptions/exception-model.md]: Use "// Contract implementation" as comment

**Review by rpereira-tae — CHANGES_REQUESTED** [review]

File: versioned_docs/version-0.1.0/site/site-model.md (hunk 8 above)
Comment (AlexisTAE) [reply versioned_docs/version-0.1.0/site/site-model.md:42]: @rpereira-tae dont understand this one

Comment (rpereira-tae) [reply versioned_docs/version-0.1.0/site/site-model.md:42]: Add a new line, a break line, you can use `<br />`

File: versioned_docs/version-0.1.0/notifications/notifications-mail.md [hunk 13]
```diff
@@ -0,0 +1,24 @@
...
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/notifications/notifications-mail.md]: # Mail configuration

MoonGuard uses the mail configuration from your Laravel app to send any email notification. 

```php
MAIL_MAILER=smtp
MAIL_HOST=mailpit
MAIL_PORT=1025
MAIL_USERNAME=null
MAIL_PASSWORD=null
MAIL_ENCRYPTION=null
MAIL_FROM_ADDRESS="hello@example.com"
MAIL_FROM_NAME="${APP_NAME}"
```

After properly setting up the parameters of your mail provider, MoonGuard is able to send email notifications.

File: versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md [hunk 14]
```diff
@@ -0,0 +1,66 @@
...
+| `saveCertificate(SslCertificate $certificate, Url $url)` | void | Saves the result of a successful SSL certificate check. This method sets the status of the SSL certificate check to "VALID" if the certificate is valid, or "INVALID" if the certificate is invalid. It also saves the expiration date and issuer of the certificate. |
+| `saveError(Exception $exception)` | void | Saves the result of a failed SSL certificate check. This method sets the status of the SSL certificate check to "INVALID", and saves the failure reason. |
+| `certificateIsValid()` | bool | Returns a boolean indicating whether the SSL certificate is valid. |
+|`certificateIsInvalid()` | bool | Returns a boolean indicating whether the SSL certificate is invalid. |
+| `certificateIsAboutToExpire(int $maxDaysToExpire)` | bool | Returns a boolean indicating whether the SSL certificate is about to expire within the specified number of days. |
+| `isEnabled()` | Attribute | Returns an `Attribute` instance that indicates whether the SSL certificate check is enabled. |
+
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md:30]: Add break line.

File: versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md [hunk 15]
```diff
@@ -0,0 +1,66 @@
...
+## Create a custom SslCertificateCheck Model
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/ssl-certificate/ssl-certificate-model.md]: This title must be the same in the model sections. 

Use:
"Using a custom x Model"

File: versioned_docs/version-0.1.0/uptime/uptime-check-model.md [hunk 16]
```diff
@@ -0,0 +1,73 @@
...
+| `saveSuccessfulCheck(Response $response)` | void | Saves the result of a successful uptime check. This method sets the status of the uptime check to "UP", updates the last check date, and saves the request duration. |
+| `saveSuccessfulCheck(Response $response)` | void | Saves the result of a failed uptime check. This method sets the status of the uptime check to "DOWN", increments the number of times the check has failed in a row, updates the last check date, saves the failure reason, and sets the request duration to null. |
+| `requestTookTooLong()` | bool | Returns a boolean indicating whether the last request took longer than the maximum allowed duration for the site. |
+| `wasFailing()` | Attribute | Returns an `Attribute` instance that indicates whether the uptime check was failing. |
+| `isEnabled()` | Attribute | Returns an `Attribute` instance that indicates whether the uptime check is enabled. |
+| `booted()` | void | Overrides the `booted` method to add a callback that sets the status last change date when the status of the uptime check changes. |
+
+# Creating a custom UptimeCheck model
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/uptime/uptime-check-model.md]: This title must be the same in the model sections. 

Use:
"Using a custom x Model"

File: versioned_docs/version-0.1.0/uptime/uptime-check-model.md [hunk 17]
```diff
@@ -0,0 +1,73 @@
...
+| `saveSuccessfulCheck(Response $response)` | void | Saves the result of a successful uptime check. This method sets the status of the uptime check to "UP", updates the last check date, and saves the request duration. |
+| `saveSuccessfulCheck(Response $response)` | void | Saves the result of a failed uptime check. This method sets the status of the uptime check to "DOWN", increments the number of times the check has failed in a row, updates the last check date, saves the failure reason, and sets the request duration to null. |
+| `requestTookTooLong()` | bool | Returns a boolean indicating whether the last request took longer than the maximum allowed duration for the site. |
+| `wasFailing()` | Attribute | Returns an `Attribute` instance that indicates whether the uptime check was failing. |
+| `isEnabled()` | Attribute | Returns an `Attribute` instance that indicates whether the uptime check is enabled. |
+| `booted()` | void | Overrides the `booted` method to add a callback that sets the status last change date when the status of the uptime check changes. |
+
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/uptime/uptime-check-model.md:39]: Add break line

File: versioned_docs/version-0.1.0/exceptions/exception-model.md [hunk 18]
```diff
@@ -0,0 +1,115 @@
...
+| `last_seen` | The date and time when the exception was last seen, casted to an immutable datetime object |
+
+### Methods
+
+| Method Name | Return Type | Description |
+| --- | --- | --- |
+| `exceptionLogs()` | HasMany | Returns a `HasMany` relationship between the `ExceptionLogGroup` model and the `ExceptionLog` model. This method allows you to retrieve the exception logs associated with this exception log group. |
+| `site()` | BelongsTo | Returns a `BelongsTo` relationship between the `ExceptionLogGroup` model and the `Site` model. This method allows you to retrieve the site associated with this exception log group. |
+
+# Create a custom Exception Log Model
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/exceptions/exception-model.md]: This title must be the same in the model sections. 

Use:
"Using a custom x Model"

File: versioned_docs/version-0.1.0/exceptions/exception-model.md [hunk 19]
```diff
@@ -0,0 +1,115 @@
...
+| `first_seen` | The date and time when the exception was first seen, casted to an immutable datetime object |
+| `last_seen` | The date and time when the exception was last seen, casted to an immutable datetime object |
+
+### Methods
+
+| Method Name | Return Type | Description |
+| --- | --- | --- |
+| `exceptionLogs()` | HasMany | Returns a `HasMany` relationship between the `ExceptionLogGroup` model and the `ExceptionLog` model. This method allows you to retrieve the exception logs associated with this exception log group. |
+| `site()` | BelongsTo | Returns a `BelongsTo` relationship between the `ExceptionLogGroup` model and the `Site` model. This method allows you to retrieve the site associated with this exception log group. |
+
```
Comment (rpereira-tae) [inline versioned_docs/version-0.1.0/exceptions/exception-model.md:71]: Add break line

**Review by rpereira-tae — CHANGES_REQUESTED** [review]

**Review by rpereira-tae — APPROVED** [review]

**Pull request merged**